service/history/vclock/vclock.go

```diff
@@ -54,14 +56,16 @@ func HappensBefore(
 }
 
 func Compare(
-	clock1 *clockspb.ShardClock,
-	clock2 *clockspb.ShardClock,
+	clock1 *clockspb.Clock,
+	clock2 *clockspb.Clock,
 ) (int, error) {
-	if clock1.GetId() != clock2.GetId() {
+	if clock1.GetClusterId() != clock2.GetClusterId() || clock1.GetShardId() != clock2.GetShardId() {
```

Comment:
We probably need to differentiate those two cases? ShardID not equal is unexpected, while clusterID not equal is expected in some cases (due to auto-forwarding).

Reply:
good point, actually even without the cluster ID, clocks generated by different cluster cannot be compared